CardsCashCasino/Data/HoldEmUtil.cs

```diff
@@ -0,0 +1,335 @@
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a straight flush, false otherwise.</returns>
+        private static bool IsStraightFlush(List<Card> cards)
+        {
+            return IsFlush(cards) && IsStraight(cards);
+        }
+
+        /// <summary>
+        /// Function to determine if a poker hand is a four of a kind.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a four of a kind, false otherwise.</returns>
+        private static bool IsFourOfAKind(List<Card> cards)
+        {
+            return cards[0].SameVal(cards[3]);
+        }
+
+        /// <summary>
+        /// Function to determine if a poker hand is a full house.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a full house, false otherwise.</returns>
+        private static bool IsFullHouse(List<Card> cards)
+        {
+            if (!cards[0].SameVal(cards[1]) || !cards[3].SameVal(cards[4]))
+            {
+                return false;
+            }
+
+            if (cards[2].SameVal(cards[1]) || cards[2].SameVal(cards[3]))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Function to determine if a poker hand is a straight.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a straight, false otherwise.</returns>
+        private static bool IsStraight(List<Card> cards)
+        {
+            // Special case: 2345A
+            if (cards[0].GetPokerValue() == 2 && cards[1].GetPokerValue() == 3 && cards[2].GetPokerValue() == 4 && cards[3].GetPokerValue() == 5 && cards[4].IsAce())
+            {
+                return true;
+            }
+
+            for (int i = 1; i < cards.Count; i++)
+            {
+                if (cards[i].GetPokerValue() != cards[i - 1].GetPokerValue() + 1)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Function to determine if a poker hand is a three of a kind.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a three of a kind, false otherwise.</returns>
+        private static bool IsThreeOfAKind(List<Card> cards)
+        {
+            return cards[2].SameVal(cards[0]) || cards[2].SameVal(cards[4]);
+        }
+
+        /// <summary>
+        /// Function to determine if a poker hand is a two pair.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a two pair, false otherwise.</returns>
+        private static bool IsTwoPair(List<Card> cards)
+        {
+            int pairs = 0;
+            for (int i = 1; i < cards.Count; i++)
```

Comment:
since we are using the same loop here to determine the number of pairs. Lets make a single, helper method that called `GetNumberOfPairs` and then store that value, then just return if it is equal to 2 or equal to 1.

This would save on computation time, which will definitely add up since we have to compare every single permutation of cards in the hand community set.